File: examples/express/historico/funko.js
const asyncHandler = require("express-async-handler");
const express = require("express");
const { Client } = require("pg");
const fs = require("node:fs");
const app = express();
const port = 3000;

USER = fs.readFileSync("/secrets/user", "utf8").trim(),
PASS = fs.readFileSync("/secrets/pass", "utf8").trim(),

app.get(
  "/",
  asyncHandler(async (req, res) => {
    names = req.query.names;
    if (!names) {
      res.send("No names provided");
      return;
    }
    names = names.split(",").map((n) => n.trim());

    const client = new Client({
      user: USER,
      password: PASS,
      host: "database",
      database: "nombres",
    });

    var response = [];
    response.push(["Año", ...names]);
    for (year = 1922; year < 2016; year++) {
      row = [year, ...names.map((n) => 0)];
      response.push(row);
    }

    await client.connect();
    for (const [i, name] of names.entries()) {
      console.log(i, name)
      const result = await client.query(
        "SELECT anio,contador FROM nombres where nombre = $1",
        [name]
      );
      console.log("filling response");
      for (row of result.rows) {
        response[row.anio - 1921][i + 1] = row.contador;
      }
    }
    console.log("responding");
    res.json(response);
  })
);

app.get("/ping", (req, res) => {
  res.send("OK");
});

app.use(express.static("public"));

app.listen(port, () => {
  console.log(`Example funko listening on port ${port}`);
});

File: examples/flask/historico/funko.py
import json
import unicodedata

import psycopg2
from flask import Flask, request

app = Flask("historico")

USER = open("/secrets/user").read().strip()
PASS = open("/secrets/pass").read().strip()
conn = psycopg2.connect(dbname="nombres", user=USER, password=PASS, host="database")


@app.route("/", methods=["GET"])
def handle():
    names = [n.strip() for n in request.args.get("names").split(",")][:4]
    cursor = conn.cursor()
    # Prepare results table
    results = [["Año"] + names]
    results += [[year] + [0 for _ in names] for year in range(1922, 2016)]
    for i, name in enumerate(names):
        nfkd_form = unicodedata.normalize("NFKD", name)
        name = "".join([c for c in nfkd_form if not unicodedata.combining(c)])
        cursor.execute(
            "SELECT anio, contador FROM nombres WHERE nombre = %s",
            (name,),
        )
        for anio, contador in cursor.fetchall():
            results[anio - 1921][i + 1] = contador
    cursor.close()
    return json.dumps(results)


@app.route("/ping")
def ping():
    cursor = conn.cursor()
    cursor.execute("SELECT 42")
    cursor.fetchall()
    cursor.close()
    return "OK"
